File: pages/cgi/printenv.pl
#!/bin/perl
print "Content-type: text/html\n\n";
print "<html>\n";
print "<title>Server-provided Environment variables</title>";
print "<body>\n";
print "<table>\n";
print "<th><td colspan=2 align=center>Environment Variables</td></th>\n";
foreach my $x (keys %ENV) {
    print "<tr><td>$x</td><td>$ENV{$x}</td></tr>\n";
}
print "</table></body></html>\n";

File: pages/cgi/printenv.js
let headers = [];
let body;

body = "<html>";
body += "<title>Server-provided Environment variables</title>";
body += "<body>";
body += "<table>";
body += "<th><td colspan=2 align=center>Environment Variables</td></th>";
for (let key in process.env) {
    body += `<tr><td>${key}</td><td>${process.env[key]}</td></tr>`;
}
body += "</table></body></html>";

headers.push("Content-Type: text/html");
headers.push(`Content-Length: ${body.length}`);

headers.forEach(header => {
    console.log(header);
});
console.log();
console.log(body);